tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/otextension/BristolOtReceiver.java

```diff
@@ -0,0 +1,134 @@
+package dk.alexandra.fresco.tools.ot.otextension;
+
+import dk.alexandra.fresco.framework.MaliciousException;
+import dk.alexandra.fresco.framework.network.Network;
+import dk.alexandra.fresco.framework.util.AesCtrDrbg;
+import dk.alexandra.fresco.framework.util.ByteArrayHelper;
+import dk.alexandra.fresco.framework.util.Drbg;
+import dk.alexandra.fresco.framework.util.StrictBitVector;
+
+import java.util.List;
+
+/**
+ * Protocol class for the party acting as the receiver in an OT extension. The
+ * functionality works by doing a batch of random OTs the first time the receive
+ * command is called and storing these results internally. When it runs out, it
+ * automatically processes a new batch. These random OTs are adjusted to work as
+ * chosen bit/message 1-out-of-2 OTs.
+ */
+public class BristolOtReceiver {
+  private final RotReceiver receiver;
+  private final OtExtensionResourcePool resources;
+  private final Network network;
+  private final int batchSize;
+
+  // The random messages received from the batched random 1-out-of-2 OTs
+  private List<StrictBitVector> randomMessages;
+  // The random choices from the batched random 1-out-of-2 OTs
+  private StrictBitVector choices;
+  // Index of the current random OT to use
+  private int offset = -1;
+
+  /**
+   * Initializes the underlying Rot functionality using {@code rotReceiver}. It
+   * will then construct OTs in batches of {@code batchSize}.
+   *
+   * @param rotReceiver
+   *          The random OT receiver instance to use
+   * @param resources
+   *          The resource pool for this specific instance
+   * @param network
+   *          The network to use
+   * @param batchSize
+   *          Size of the OT extension batch the protocol will construct
+   */
+  public BristolOtReceiver(RotReceiver rotReceiver,
+      OtExtensionResourcePool resources, Network network, int batchSize) {
+    this.receiver = rotReceiver;
+    this.resources = resources;
+    this.network = network;
+    this.batchSize = batchSize;
+  }
+
+  /**
+   * Receive the serialized message from the current 1-out-of-2 OT.
+   *
+   * @param choiceBit
+   *          Choice-bit. False for message 0, true for message 1.
+   * @return The serialized message from the OT
+   */
+  public byte[] receive(Boolean choiceBit) {
```

Comment:
Is there a reason `choiceBit` should not be a primitive type `boolean` instead of an object `Boolean`?

Reply:
Changed in 6bd02da5